File: test/Integration/test-java/src/main/java/com/function/TableNotPresentTrigger.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.function;

import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.sql.annotation.SQLTrigger;
import com.function.Common.SqlChangeProduct;


public class TableNotPresentTrigger {
    @FunctionName("TableNotPresentTrigger")
    public void run(
            @SQLTrigger(
                name = "changes",
                tableName = "[dbo].[TableNotPresent]",
                connectionStringSetting = "SqlConnectionString")
                SqlChangeProduct[] changes,
            ExecutionContext context) {

        throw new RuntimeException("Associated test case should fail before the function is invoked.");
    }
}

File: test-outofproc/TableNotPresentTrigger.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System;
using System.Collections.Generic;
using DotnetIsolatedTests.Common;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Extensions.Sql;

namespace DotnetIsolatedTests
{
    public static class TableNotPresentTrigger
    {
        /// <summary>
        /// Used in verification of the error message when the user table is not present in the database.
        /// </summary>
        [Function(nameof(TableNotPresentTrigger))]
        public static void Run(
            [SqlTrigger("[dbo].[TableNotPresent]", "SqlConnectionString")]
            IReadOnlyList<SqlChange<Product>> products)
        {
            throw new NotImplementedException("Associated test case should fail before the function is invoked.");
        }
    }
}
